docs/tools/testing/unit_testing.md

```diff
@@ -1,14 +1,104 @@
 ## Unit testing
 
-We perform integration and system testing in an environment where we run the tests with the full Mbed OS. Other testing tools for Mbed OS require specific embedded hardware, which means traditional unit testing is not possible.
+This document describes how to run and write unit tests for Mbed OS. In case of running into problems see [troubleshooting](#troubleshooting) section.
+
+### Introduction
+
+We perform integration and system testing in an environment where we run the tests with the full Mbed OS. Other testing tools for Mbed OS require specific embedded hardware, which means actual unit testing is not possible.
 
 Unit testing takes place in a build environment where we test each C or C++ class or module in isolation. This means we build test suites into separate test binaries and stub all access outside to remove dependencies on any specific embedded hardware or software combination. This allows us to complete the testing using native compilers on the build machine.
 
+### Prerequisites
+
+* GNU toolchains installed.
```

Comment:
Query: How often do these versions change? It may be better to link elsewhere than for our engineers in charge of this to remember to inform our docs team that the versions have changed.